client/src/components/subscription/subscriptionForm.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+
+/** TO DO
+ * Indicator if the form is correctly or incorrectly filled out. (done in dev, need to be done in prod)
+ * We don't need to require the userId as state. This should be filled in auto because route for creating is /user/:userUUID/subscriptions (done)
+ * After creating a subscription, we need to reset all the input values to normal. (done)
+*/
+
+class SubscriptionForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // When a field is empty, and not required, it should save as NULL instead of an empty string.
+    this.state = {
+      name: '',
+      nickname: '',
+      due_date: '',
+      reminder_days: '',
+      amount: '',
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  };
+
+  handleChange(e, key) {
+    e.preventDefault();
+    this.setState({ [key]: e.target.value });
+  }
+
+  async handleSubmit(e) {
+    e.preventDefault();
+
+    const subscriptionInfo  = this.state;
+    const { performUpdate } = this.props;
+    const httpMethod = performUpdate ? 'PATCH' : 'PUT';
+
+    /* If parent component passed this.props.performUpdate
+       then we want to iterate over performUpdate and place all the values that aren't updating into 
+       subscriptionInfo, which we will pass to our request.
+    */
+    if (performUpdate) {
+      Object.keys(performUpdate).forEach(key => {
+        if (subscriptionInfo[key] === '') {
+          subscriptionInfo[key] = performUpdate[key];
+        }
+      });
+
+      subscriptionInfo.uuid = performUpdate.subscription_uuid;
+    }
+
+    // User wants to update their subscription.
+    // perform update contains the subscription row, including subscription_uuid.
+
+    try {
+      await fetch(`${window.location.pathname}/subscriptions`, {
+        method: httpMethod,
+        headers: {
+          'Content-Type': 'application/json'
+        },
```

Comment:
can this just be one line?

```suggestion
        headers: { 'Content-Type': 'application/json' },
```